src/lib.rs

```diff
@@ -45,6 +45,9 @@ pub fn run_program_with_custom_state(bin_path: &str, mut vm: VMState) -> (U256,
         let opcode = vm.get_opcode(&opcode_table);
         if vm.predicate_holds(&opcode.predicate) {
             match opcode.variant {
+                // TODO: Properly handle what happens
+                // when the VM runs out of ergs/gas.
+                _ if vm.gas_left() == 0 => break,
```

Comment:
[Maybe `panic` instead of `break`ing.](https://github.com/matter-labs/zksync-era/blob/main/docs/specs/zk_evm/vm_specification/zkSync_era_virtual_machine_primer.md#basic-instructions:~:text=Instructions%20are%20executed%20one%20after%20another%2C%20and%20every%20instruction%20has%20a%20gas%20cost%20measured%20in%20gas.%20A%20program%20that%20runs%20out%20of%20gas%20panics%20and%20none%20of%20its%20side%20effects%20are%20performed.)

Reply:
I get what you mean, but panic-ing in Rust is not the same as panic-ing in the VM, the VM just reverts state, Rust kills the thread, so I think breaking is fine for now.